lmsweb/registartion.py

```diff
@@ -0,0 +1,114 @@
+import csv
+import logging
+import os
+import typing
+
+from lms.lmsweb import config
+from lms.lmsweb import models
+
+import requests
+
+
+_logger = logging.getLogger(__name__)
+
+
+class UserToCreate(typing.NamedTuple):
+    first_name: str
+    last_name: str
+    email: str
+    password: str
+
+    def to_dict(self):
+        return self._asdict()
+
+    @classmethod
+    def get_fields(cls):
+        return cls._fields
+
+
+class UserRegistrationCreator(object):
+    _session = requests.Session()
+
+    def __init__(self, users_to_create: typing.Sequence[UserToCreate]):
+        self._users_to_create = users_to_create
+        self._failed_users: typing.List[UserToCreate] = []
+
+    @classmethod
+    def from_csv_file(cls, file_path: str) -> 'UserRegistrationCreator':
+        """
+        CSV file should be with three columns,
+        and in the header: first_name,last_name,email and password[optional]
+        """
+
+        if not os.path.exists(file_path):
+            raise ValueError
+
+        with open(file_path, 'r') as file_reader:
+            csv_records = csv.DictReader(file_reader)
+
+        users = []
+        for record in csv_records:
+            if 'password' not in record:
+                record['password'] = os.urandom(10).hex()
```

Comment:
Const & a bit longer (12)? 'cause it's an hash.
Even better:

```
import string
import random

def generate_pass(length)
    ''.join(random.choices(string.printables.strip(), k=length))
````

Reply:
Done